src/components/ConfigEditor.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { ChangeEvent } from 'react';
+import { InlineField, Input, SecretInput } from '@grafana/ui';
+import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
+import { MyDataSourceOptions, MySecureJsonData } from '../types';
+
+interface Props extends DataSourcePluginOptionsEditorProps<MyDataSourceOptions, MySecureJsonData> {}
+
+export function ConfigEditor(props: Props) {
+  const { onOptionsChange, options } = props;
+  const onUrlChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const jsonData = {
+      ...options.jsonData,
+      dashboardUrl: event.target.value,
+    };
+    onOptionsChange({ ...options, jsonData });
+  };
+
+  // Secure field (only sent to the backend)
+  const onAPIKeyIdChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onOptionsChange({
+      ...options,
+      secureJsonData: {
+        apiKeyId: event.target.value,
+      },
+    });
+  };
+
+  const onAPIKeyTokenChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onOptionsChange({
+      ...options,
+      secureJsonData: {
+        apiKeyToken: event.target.value,
+      },
+    });
+  };
+
+  const onResetAPIKeyId = () => {
+    onOptionsChange({
+      ...options,
+      secureJsonFields: {
+        ...options.secureJsonFields,
+        apiKeyId: false,
+      },
+      secureJsonData: {
+        ...options.secureJsonData,
+        apiKeyId: '',
+      },
+    });
+  };
+
+  const onResetAPIKeyToken = () => {
+    onOptionsChange({
+      ...options,
+      secureJsonFields: {
+        ...options.secureJsonFields,
+        apiKeyToken: false,
+      },
+      secureJsonData: {
+        ...options.secureJsonData,
+        apiKeyToken: '',
+      },
+    });
+  };
+
+  // const onResolutionChange = (event: ChangeEvent<HTMLInputElement>) => {
+  //   const jsonData = {
+  //     ...options.jsonData,
+  //     resolution: parseFloat(event.target.value),
+  //   };
+  //   onOptionsChange({ ...options, jsonData });
+  // };
+
+  const { jsonData, secureJsonFields } = options;
+  const secureJsonData = (options.secureJsonData || {}) as MySecureJsonData;
+
+  return (
+    <div className="gf-form-group">
+      <InlineField label="Dashboard URL" labelWidth={20} required>
+        <Input
+          onChange={onUrlChange}
+          value={jsonData.dashboardUrl || ''}
```

Comment:
should we just default here to https://dashboard.pepperdata.com

Reply:
I can't find the "correct" way to set defaults for these, seems like all I can do is call the `onOptionsChange` handler provided. It's a little hacky but I can add a useEffect to call the handler on load to set the default.